jax/experimental/sparse.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from functools import partial
+
+import scipy.sparse.linalg
+import jax.numpy as np
+from jax.numpy.lax_numpy import _wraps
+from .. import lax
+
+_T = lambda x: np.swapaxes(x, -1, -2)
```

Comment:
This should add a conjugate so it works properly on complex valued array. Also please test complex numbers, too.

Reply:
looks like the way i'm currently making the square matrix doesn't take into account `dtype`. Is there an alternative to using numpy's `randn`?